submariner/IPV6-OVN.md

```diff
@@ -0,0 +1,105 @@
+# **Submariner OVN CNI Enhancement for IPv6 Support**
+
+## **Summary**
+
+This proposal outlines the changes required in Submariner for  OVN Kubernetes
+CNI to enable IPv6 support, ensuring seamless connectivity between clusters
+using Submariner. The main proposal has the full design
+[IPv6 Datapath Enhancements](IPV6-datapath.md)
+This covers only the OVN CNI part of it.
+
+---
+
+## Design Details
+
+The OVNKubernetes handler programs network policies and routes to direct traffic from
+the gateway and non-gateway nodes to direct the traffic to the remote cluster.
+At present the routes are only programmed for IPv4 for addresses. We need to enhance
+this to support IPV6 addresses as well.
+
+The handler for creating the Gateway and NonGateway routes needs to be enhanced.
+
+### GatewayRoute CRD
+
+The GatewayRoute will now create a new CR for with  the IPV6 next hops and
```

Comment:
I assume we want to  create GatewayRoute,NonGatewayRoute CRs per remote endpoint IP family only if this IP family is supported by local-cluster.

Reply:
We check this at the time of join right? Do we need another check here?

Reply:
```suggestion
A GatewayRoute CR will be created with the IPV6 next hops and
```

Reply:
> We check this at the time of join right? Do we need another check here?

Yep, we verify that we can create at least one tunnel between two clusters.

Assuming the following use case:
Cluster 1 is IPV4
Cluster2 is dual stack

The validation is green for this case.
How many GatewayRoute,NonGatewayRoute CRs should be created in Cluster1 to route traffic to Cluster2?

Do you see any point in creating GatewayRoute,NonGatewayRoute for IPV6?

Reply:
Done.